### ISSUE TITLE
`uninlined_format_args` main error message doesn't take specific macro into account

### ISSUE THREAD

**Issue by Nemo157** [rust-lang/rust-clippy#9715]
### Summary

The first error message for uninlined args references the `format!` macro, even when it's not the `format!` macro that's being invoked.

### Reproducer

I tried this code:

```rust
pub fn foo(mut w: impl std::io::Write) -> std::io::Result<()> {
    let x = 5;
    write!(w, "{}", x)?;
    Ok(())
}
```

I expected to see this happen: a warning about the use of a variable that could be inlined in `write!`

Instead, this happened: a warning about the use of a variable that could be inlined in `format!`

```
warning: variables can be used directly in the `format!` string
 --> src/lib.rs:3:5
  |
3 |     write!(w, "{}", x)?;
  |     ^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#uninlined_format_args
  = note: `#[warn(clippy::uninlined_format_args)]` on by default
help: change this to
  |
3 -     write!(w, "{}", x)?;
3 +     write!(w, "{x}")?;
  |
```

### Version

```text
clippy: 0.1.66 (2022-10-24 758f196)
```

**Comment by nyurik** [comment]
This is more of a documentation issue than a bug.  The warning you see is generated at 
https://github.com/rust-lang/rust-clippy/blob/f9c1d155b4c9cad224fe96aad486993dc123c9b6/clippy_lints/src/format_args.rs#L337-L342

The `format!` does not refer to the specific macro, but instead mentions all format-using macros like `print!, write!, assert!, panic!, ...`.   Perhaps we should rephrase it some other way?
